contracts/tokens/SecurityToken.sol

```diff
@@ -540,34 +540,28 @@ contract SecurityToken is StandardToken, DetailedERC20, ReentrancyGuard, Registr
         uint256 _value,
         bytes _data,
         bool _isTransfer
-        ) internal checkGranularity(_value) returns (bool) {
+    ) internal checkGranularity(_value) returns (bool) {
         if (!transfersFrozen) {
-            if (modules[TRANSFER_KEY].length == 0) {
-                return true;
-            }
             bool isInvalid = false;
             bool isValid = false;
-            bool isForceValid = false;
             bool unarchived = false;
             address module;
-            for (uint8 i = 0; i < modules[TRANSFER_KEY].length; i++) {
+            for (uint256 i = 0; i < modules[TRANSFER_KEY].length; i++) {
                 module = modules[TRANSFER_KEY][i];
                 if (!modulesToData[module].isArchived) {
                     unarchived = true;
                     ITransferManager.Result valid = ITransferManager(module).verifyTransfer(_from, _to, _value, _data, _isTransfer);
                     if (valid == ITransferManager.Result.INVALID) {
                         isInvalid = true;
-                    }
-                    if (valid == ITransferManager.Result.VALID) {
+                    } else if (valid == ITransferManager.Result.VALID) {
                         isValid = true;
-                    }
-                    if (valid == ITransferManager.Result.FORCE_VALID) {
-                        isForceValid = true;
+                    } else if (valid == ITransferManager.Result.FORCE_VALID) {
+                        return true;
                     }
```

Comment:
@maxsam4 I don't think we can do this as we need to guarantee that all TMs see all transfers (e.g. in case they are maintaining internal state related to the transfer). If _isTransfer == false we could bypass, but I think it is cleaner to just ensure that all TMs see all transfers rather than breaking out of the loop by returning `true` in the FORCE_VALID case.

Reply:
Yeah, that makes sense. Gonna push a fix in a bit. So basically, the only optimization left is using `else if` rather than `if`.